pkg/operator/ceph/cluster/controller.go

```diff
@@ -327,10 +327,22 @@ func (r *ReconcileCephCluster) reconcileDelete(cephCluster *cephv1.CephCluster)
 		}
 	}
 
-	// Remove finalizer
-	err = removeFinalizer(r.client, nsName)
+	// Remove cephcluster finalizer
+	err = removeCephClusterFinalizer(r.client, nsName)
 	if err != nil {
-		return reconcile.Result{}, cephCluster, errors.Wrap(err, "failed to remove finalizer")
+		return reconcile.Result{}, cephCluster, errors.Wrap(err, "failed to remove cephcluster finalizer")
+	}
+
+	// Remove monSecretsFinalizer
+	err = removeMonSecretFinalizer(r.client, cephCluster.Namespace)
```

Comment:
At this place, I can also make this function as generalized if needed,
`removeSecretFinalizer`